src/controllers/monster.js

```diff
@@ -602,6 +603,95 @@ class Monster extends Controller {
+							function destinationPoint(distance, bearing) {
+								// sinφ2 = sinφ1⋅cosδ + cosφ1⋅sinδ⋅cosθ
+								// tanΔλ = sinθ⋅sinδ⋅cosφ1 / cosδ−sinφ1⋅sinφ2 / this should be memorized as common knowledge
+								const δ = distance / earthRadius; // angular distance in radians
+								const θ = degreesToRadians(bearing);
+
+								const φ1 = degreesToRadians(data.latitude), λ1 = degreesToRadians(data.longitude);
+
+								const sinφ2 = Math.sin(φ1) * Math.cos(δ) + Math.cos(φ1) * Math.sin(δ) * Math.cos(θ);
+								const φ2 = Math.asin(sinφ2);
+								const y = Math.sin(θ) * Math.sin(δ) * Math.cos(φ1);
+								const x = Math.cos(δ) - Math.sin(φ1) * sinφ2;
+								const λ2 = λ1 + Math.atan2(y, x);
+
+								const lat2 = radiansToDegrees(φ2);
+								const lon2 = radiansToDegrees(λ2);
+								return [lat2,lon2];
+							}
+
+							const distance = 0.070 * Math.sqrt(3.0);
+							for (var i=0; i<6; i++) {
+								let bearing = i*60.0;
+								let pt2 = destinationPoint(distance,bearing);
+								coords.push(pt2)
+							}						
+						}
+						
+						const headers = {
+								'Content-Type': 'application/json',
+								'X-Dragonite-Admin-Secret': this.config.scouts?.dragoniteSecret
+								}
+						const url = this.config.scouts?.scoutURL
+						this.log.info(`[SCOUT] ${data.encounter_id}: ${monster.name} posting to ${url} with ${coords.map(([lat,lon])=>`[${lat.toFixed(3)},${lon.toFixed(3)}]`).join(', ')}`)
+						
+						try {
+							const hrstart = process.hrtime()
+							const timeoutMs = this.config.scouts?.dragoniteTimeout || 10000
+							const source = axios.CancelToken.source()
+							const timeout = setTimeout(() => {
+								source.cancel(`[SCOUT] Timeout waiting for scout response - ${timeoutMs}ms`)
+								// Timeout Logic
+							}, timeoutMs)
+
+							const result = await axios.post(url, coords, {cancelToken: source.token}, {headers: headers})
+							clearTimeout(timeout)
+							if (result.status !== 200) {
+								this.log.warn(`[SCOUT] Failed Scout Attempt: Got ${result.status}. Error: ${result.data ? result.data.reason : '?'}.`)
+								return null
+							}
+							const hrend = process.hrtime(hrstart)
+							const hrendms = hrend[1] / 1000000
+
+							if (result.data.includes('<')) { // check for HTML error response
+								this.log.warn(`[SCOUT] Failed Scout Attempt: Got invalid response from Dragonite - ${result.data}`)
+								return null
+							}
+							data.scoutResult = data.scoutResult + result.data
+							this.log.info(`[SCOUT] ${data.encounter_id}: ${monster.name} Scout sent successfully!`)
+							
+						} catch (error) {
+							if (error.response) {
+								this.log.warn(`[SCOUT] Scout Error - Got ${error.response.status}. Error: ${error.response.data ? error.response.data.reason : '?'}.`)
+							} else {
+								this.log.warn(`[SCOUT] Scout NoResponse Error: ${error}.`)
+							}
+							return null
```

Comment:
We shouldn't be returning null in any of these error conditions - these will be abandoning the whole monster processing in case of a scout error?

Reply:
I've removed the returns and used a flag instead.